shelley/chain-and-ledger/executable-spec/test/Generator/Update/QuickCheck.hs

```diff
@@ -1,28 +1,50 @@
 {-# LANGUAGE DataKinds #-}
+{-# LANGUAGE FlexibleContexts #-}
 {-# LANGUAGE FlexibleInstances #-}
 {-# LANGUAGE PatternSynonyms #-}
 {-# LANGUAGE TupleSections #-}
 {-# LANGUAGE TypeSynonymInstances #-}
 {-# LANGUAGE UndecidableInstances #-}
 
 module Generator.Update.QuickCheck
-  ( genPParams )
+  ( genPParams
+  , genUpdate
+  )
   where
 
+import           Control.Monad(join)
 import           Data.Ratio ((%))
+import           Data.Set (Set)
+import qualified Data.Set   as Set (fromList)
+import qualified Data.Map.Strict as Map (fromList)
+import qualified Data.ByteString.Char8 as BS (pack)
 
 import           Test.QuickCheck (Gen)
 import qualified Test.QuickCheck as QC
 
 import           BaseTypes (Nonce (NeutralNonce), UnitInterval, mkNonce)
+import           Cardano.Crypto.Hash (HashAlgorithm)
+import           Cardano.Ledger.Shelley.Crypto (HASH)
 import           Coin (Coin (..))
 import           Examples (unsafeMkUnitInterval)
+import           Keys (hash)
 import           Generator.Core.QuickCheck (genInteger, genNatural, genWord64,
                      increasingProbabilityAt)
 import           Numeric.Natural (Natural)
 import           PParams (PParams (..))
 import           Slot (EpochNo (EpochNo))
-
+import           Updates (Ppm (..)
+                         ,PParamsUpdate(..)
+                         ,PPUpdate(..)
+                         ,SystemTag(..)
+                         ,InstallerHash(..)
+                         ,ApVer(..)
+                         ,ApName(..)
+                         ,Applications(..)
+                         ,Mdt(..)
+                         ,AVUpdate(..)
+                         ,Update(..))
```

Comment:
please keep to the formatting used, i.e., multiple lines for imports. I thought that stylish does this automatically ?

Reply:
@mgudemann I followed the best practice that attempts to minimize diffs in a frequent case when there are many imports, but many of them change.

Do you think we should prefer following `stylish` at all times instead?

Reply:
my opinion here is yes, follow automated formatting using stylish, I myself find it easy to skip import changes and redundant or missing imports are caught anyhow using CI